exchanges/deribit/deribit_wrapper.go

```diff
@@ -0,0 +1,1276 @@
+		// TODO: candlestick data for asset item option_combo, future_combo, and option not supported yet
+	}
+	return nil, fmt.Errorf("%w candlestick data for asset type %v", asset.ErrNotSupported, a)
+}
+
+// GetHistoricCandlesExtended returns candles between a time period for a set time interval
+func (d *Deribit) GetHistoricCandlesExtended(ctx context.Context, pair currency.Pair, a asset.Item, interval kline.Interval, start, end time.Time) (*kline.Item, error) {
+	req, err := d.GetKlineExtendedRequest(pair, a, interval, start, end)
+	if err != nil {
+		return nil, err
+	}
+	var tradingViewData *TVChartData
+	timeSeries := make([]kline.Candle, 0, req.Size())
+	switch a {
+	case asset.Futures, asset.Spot:
+		for x := range req.RangeHolder.Ranges {
+			intervalString, err := d.GetResolutionFromInterval(interval)
+			if err != nil {
+				return nil, err
+			}
+			if d.Websocket.IsConnected() {
+				tradingViewData, err = d.WSRetrievesTradingViewChartData(d.formatFuturesTradablePair(req.RequestFormatted), intervalString, req.RangeHolder.Ranges[x].Start.Time, req.RangeHolder.Ranges[x].End.Time)
+			} else {
+				tradingViewData, err = d.GetTradingViewChart(ctx, d.formatFuturesTradablePair(req.RequestFormatted), intervalString, req.RangeHolder.Ranges[x].Start.Time, req.RangeHolder.Ranges[x].End.Time)
+			}
+			if err != nil {
+				return nil, err
+			}
+			checkLen := len(tradingViewData.Ticks)
+			if len(tradingViewData.Open) != checkLen ||
+				len(tradingViewData.High) != checkLen ||
+				len(tradingViewData.Low) != checkLen ||
+				len(tradingViewData.Close) != checkLen ||
+				len(tradingViewData.Volume) != checkLen {
+				return nil, fmt.Errorf("%s - %v: invalid trading view chart data received", a, d.formatFuturesTradablePair(req.RequestFormatted))
+			}
+			for x := range tradingViewData.Ticks {
+				timeSeries = append(timeSeries, kline.Candle{
+					Open:   tradingViewData.Open[x],
+					High:   tradingViewData.High[x],
+					Low:    tradingViewData.Low[x],
+					Close:  tradingViewData.Close[x],
+					Volume: tradingViewData.Volume[x],
+					Time:   time.UnixMilli(tradingViewData.Ticks[x]),
+				})
+			}
+		}
+		return req.ProcessResponse(timeSeries)
+	case asset.OptionCombo, asset.FutureCombo, asset.Options:
+		// TODO: candlestick data for asset item option_combo, future_combo, and option not supported yet
+	}
+	return nil, fmt.Errorf("%w candlestick data for asset type %v", asset.ErrNotSupported, a)
+}
+
+// GetServerTime returns the current exchange server time.
+func (d *Deribit) GetServerTime(ctx context.Context, _ asset.Item) (time.Time, error) {
+	return d.GetTime(ctx)
+}
+
+// AuthenticateWebsocket sends an authentication message to the websocket
+func (d *Deribit) AuthenticateWebsocket(ctx context.Context) error {
+	return d.wsLogin(ctx)
+}
+
+// GetFuturesContractDetails returns all contracts from the exchange by asset type
+func (d *Deribit) GetFuturesContractDetails(context.Context, asset.Item) ([]futures.Contract, error) {
```

Comment:
This is supported and you should use https://docs.deribit.com/#public-get_instruments endpoint to populate contract information